File: scripts/slice_data.pl
#! /usr/bin/env perl

# Purpose: Create a file with data point between start_date and end_date, inclusive
# Date format is YYYYMM
#
# Usage: slice_data.pl start_date end_date < file_with_data.dat > sliced_data_file.dat
#
use strict;

main();

sub main {

    $| = 1;
    
    my $start_date = $ARGV[0] || die "Must provide start date";
    my $end_date   = $ARGV[1] || die "Must provide end date";

    my @headers = ();
    my @lines = ();

    LINE:
    while( <STDIN> ) {

	chomp;
	my @fields = split ' ',$_;
	
	if ($fields[1] eq 'gvkey') {
	    @headers = @fields;
	    next LINE;
	}

	my $date = $fields[0];

	if (($date ge $start_date) && ($date le $end_date)) {
	    push @lines, \@fields;
	}

    }
    print(join(' ',@headers),"\n") if scalar(@headers);

    foreach my $line (@lines) {
	
	print(join(' ',@$line),"\n");

    }

}

File: scripts/slice_data.py
#! /usr/bin/env python3

#
# usage: slice_data.py start_date end_data < in-file.dat > out-file.dat


from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import time
import os
import sys

def main():
    assert(len(sys.argv)>2)
    start_date = sys.argv[1]
    end_date  = sys.argv[2]

    for line in sys.stdin:
        line = line.strip()
        fields = line.split(' ')
        date = fields[0]
        if (date >= start_date) and (date <= end_date):
            print(line)

if __name__ == "__main__":
    main()
